src/domain/event/route/EventRoute.tsx

```diff
@@ -1,10 +1,46 @@
 import React, { FunctionComponent } from 'react';
-import { Switch } from 'react-router-dom';
+import { Switch, useHistory, useParams } from 'react-router-dom';
 import { useTranslation } from 'react-i18next';
+import { useQuery } from '@apollo/client';
 
 import { getCurrentLanguage } from '../../../common/translation/TranslationUtils';
 import AppRoute from '../../app/AppRoute';
 import Event from '../Event';
+import { eventQuery as EventQueryType } from '../../api/generatedTypes/eventQuery';
+import eventQuery from '../queries/eventQuery';
+
+export const useEventRouteGoBack = () => {
+  const { i18n } = useTranslation();
+  const history = useHistory();
+  const currentLocale = getCurrentLanguage(i18n);
+  const { childId, eventGroupId, eventId } =
+    useParams<{ childId: string; eventGroupId: string; eventId: string }>();
+
+  const { data: eventData } = useQuery<EventQueryType>(eventQuery, {
+    variables: { id: eventId, childId },
+    skip: !!eventGroupId || !eventId,
+  });
+
+  // Go to event group page
+  if (eventGroupId) {
+    return () =>
+      history.push(
+        `/${currentLocale}/profile/child/${childId}/event-group/${eventGroupId}`
+      );
+  }
```

Comment:
If eventGroupId exists, it basically means that we are on event group page? Maybe it could be communicated better by assigning !!eventGroupId to a variable. It is not very clear immidiately that `!!eventGroupId` means that we are on................................................. Actually I realized that I cant find a way to get eventGroupId defined, on what page exactly is this used?

Reply:
To me this solution looks ok but I think that it just needs little bit code polishing for readability. For example when reading `!!eventGroupId || !eventId` I have hard time figuring out the actual situation when query is skipped.

Reply:
removed the  `if (eventGroupId) -block`. Also, now returning a string instead of void.